autorest/azure/example/main.go

```diff
@@ -244,7 +244,7 @@ func main() {
 	}
 
 	client := &autorest.Client{}
-	client.Authorizer = spt
+	client.Authorizer = autorest.NewBearerAuthorizer(spt)
```

Comment:
The Authorizer interface is dependent on autorest.PrepareDecorator.  

```Go
type Authorizer interface {
	WithAuthorization() PrepareDecorator
}
```
This will probably require more refactoring that the Token implements the interface without creating a dependency to autorest package. 

This implementation appears to me a bit more clean, because the authorizer is only responsible to build the HTTP header.